### PR TITLE
chore: cpp11 wrapper for `igraph_hcass2()`

### LINKED ISSUE #728: Convert the new `R_igraph_hcass2()` to C++
On top of #720.

### PR DESCRIPTION
**Pull request by Antonov548**
Closes #728

### CODE REVIEW

**Review by krlmlr — COMMENTED** [review]
Thanks.

- Can you keep `igraphhcass()` in the C file, and use it from there (perhaps declare with `extern "C"`)?
- Can you please rename `sample.cpp` to something more suitable?
- Can you please add EOL at the end of each file that doesn't have it yet?
- Can you please edit the description of the PR so that it contains `Closes #<issue number>` ? This closes the linked issue automatically when the PR is merged.

We can deal with the `C_` prefix after this is merged, to avoid conflicts.

**Comment by Antonov548** [comment]
> Thanks.
> 
> * Can you keep `igraphhcass()` in the C file, and use it from there (perhaps declare with `extern "C"`)?
> * Can you please rename `sample.cpp` to something more suitable?
> * Can you please add EOL at the end of each file that doesn't have it yet?
> * Can you please edit the description of the PR so that it contains `Closes #<issue number>` ? This closes the linked issue automatically when the PR is merged.
> 
> We can deal with the `C_` prefix after this is merged, to avoid conflicts.


Thanks!

I will update pull request. Seems I didn't properly setup editorconfig, now it works.

File: src/cpprinterface.cpp [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+[[cpp11::register]]
+SEXP igraph_hcass2(SEXP n, SEXP ia, SEXP ib) {
+  SEXP result;
+  int c_n = INTEGER(n)[0];
+
+  igVector a(c_n);
+  igVector b(c_n);
+
+  PROTECT(result = NEW_INTEGER(c_n));
```
Comment (krlmlr) [inline src/cpprinterface.cpp:20]: Can we use cpp11 objects here?

File: src/cpprinterface.cpp [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+#include <Rinternals.h>
+#include <Rdefines.h>
+#include <R_ext/Visibility.h>
+
+#include "igraph_vector.hpp"
+
+extern "C" int igraphhcass2(int *n, int *ia, int *ib, int *iorder, int *iia, int *iib);
+
+[[cpp11::register]]
+SEXP igraph_hcass2(SEXP n, SEXP ia, SEXP ib) {
```
Comment (krlmlr) [inline src/cpprinterface.cpp:13]: Let's use cpp11 to help us here.

```suggestion
SEXP igraph_hcass2(int c_n, SEXP ia, SEXP ib) {
```

Ideally, we could also do

```c
SEXP igraph_hcass2(int c_n, igVector a, igVector b) {
```

to harness the full power.

**Review by krlmlr — COMMENTED** [review]
Thanks! This is a case study of what a modern wrapper could look like, let's spend a bit more time here.

**Comment by krlmlr** [comment]
Thanks!

**Pull request merged**